paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethod.kt

```diff
@@ -22,61 +23,81 @@ internal enum class SupportedPaymentMethod(
     val code: String,
     @StringRes val displayNameResource: Int,
     @DrawableRes val iconResource: Int,
-    val formSpec: FormSpec?
+    val formSpec: FormSpec?,
+    val requiresMandate: Boolean,
+    val userRequestedConfirmSaveForFutureSupported: Boolean
 ) {
     Card(
         "card",
         R.string.stripe_paymentsheet_payment_method_card,
         R.drawable.stripe_ic_paymentsheet_pm_card,
-        null
+        null,
+        requiresMandate = PaymentMethod.Type.Card.requiresMandate,
```

Comment:
We previously were using `val code: String` to avoid circular dependency, but now we can replace that with a `val type: PaymentMethod.Type`, then `requiresMandate` can be a method that returns `type.requiresMandate` (or wherever you need that just call it from the `type`). There are other cases where having the `PaymentMethod.Type` as a property here makes the code cleaner, so I think we should do that.

Reply:
I would like to tackle this item in another PR: https://github.com/stripe/stripe-android/pull/4196    I am just working on it now.